pykafka/cli/kafka_tools.py

```diff
@@ -0,0 +1,285 @@
+          boundaries. If there is only one segment, as when Kafka has just
+          started, the only offsets found will be [0, <latest_offset>].
+
+    :param client: KafkaClient connected to the cluster.
+    :type client:  :class:`pykafka.KafkaClient`
+    :param topic:  Name of the topic.
+    :type topic:  :class:`str`
+    :param offset: Offset to reset to. Can be earliest, latest or a datetime.
+        Using a datetime will reset the offset to the latest message published
+        *before* the datetime.
+    :type offset: :class:`pykafka.common.OffsetType` or
+        :class:`datetime.datetime`
+    """
+    # Don't auto-create topics.
+    if topic not in client.topics:
+        raise ValueError('Topic {} does not exist.'.format(topic))
+    topic = client.topics[topic]
+
+    offsets = fetch_offsets(client, topic, offset)
+    print tabulate.tabulate(
+        [(k, v.offset[0]) for k,v in offsets.iteritems()],
+        headers=['Partition', 'Offset'],
+        numalign='center',
+    )
+
+
+def print_topics(client):
+    """Print all topics in the cluster.
+
+    :param client: KafkaClient connected to the cluster.
+    :type client:  :class:`pykafka.KafkaClient`
+    """
+    print tabulate.tabulate(
+        [(t.name,
+          len(t.partitions),
+          len(t.partitions.values()[0].replicas) - 1)
+         for t in client.topics.values()],
+        headers=['Topic', 'Partitions', 'Replication'],
+        numalign='center',
+    )
+
+
+def reset_offsets(client, topic, consumer_group, offset):
+    """Reset offset for a topic/consumer group.
+
+    NOTE: Time-based offset lookups are not precise, but are based on segment
+          boundaries. If there is only one segment, as when Kafka has just
+          started, the only offsets found will be [0, <latest_offset>].
+
+    :param client: KafkaClient connected to the cluster.
+    :type client:  :class:`pykafka.KafkaClient`
+    :param topic:  Name of the topic.
+    :type topic:  :class:`str`
+    :param consumer_group: Name of the consumer group to reset offsets for.
+    :type consumer_groups: :class:`str`
+    :param offset: Offset to reset to. Can be earliest, latest or a datetime.
+        Using a datetime will reset the offset to the latest message published
+        *before* the datetime.
+    :type offset: :class:`pykafka.common.OffsetType` or
+        :class:`datetime.datetime`
+    """
+    # Don't auto-create topics.
+    if topic not in client.topics:
+        raise ValueError('Topic {} does not exist.'.format(topic))
+    topic = client.topics[topic]
+
+    offsets = fetch_offsets(client, topic, offset)
+    consumer = topic.get_simple_consumer(consumer_group=consumer_group,
+                                         auto_start=False)
+    for owned_partition, partition in consumer._partitions.iteritems():
+        owned_partition.last_offset_consumed = int(offsets[partition.id].offset[0])
```

Comment:
@kbourgoin I realized you should at least be using `set_offset` here instead of directly modifying `last_offset_consumed`. That way, you're at least safer if the underlying `OwnedPartition` implementation changes.